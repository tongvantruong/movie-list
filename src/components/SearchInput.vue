<template>
  <div class="search-input">
    <VTextField
      test-id="search-input"
      data-cy="search-input"
      rounded
      prepend-inner-icon="mdi-magnify"
      density="comfortable"
      variant="solo"
      placeholder="Search by Title"
      hide-details
      single-line
      clearable
      @input="emit('input')"
      @click:clear="onClear"
      v-model="model"
    />
  </div>
</template>
<script setup lang="ts">
import { type ModelRef } from 'vue'

const model: ModelRef<string> = defineModel({ type: String, default: '' })
const emit = defineEmits<{
  input: []
  clear: []
}>()

function onClear() {
  model.value = ''
  emit('clear')
}
</script>

<style scoped lang="scss">
.search-input {
  width: 100%;
}
</style>
