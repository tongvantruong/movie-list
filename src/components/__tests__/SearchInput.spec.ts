import { describe, it, expect } from 'vitest'
import { createVuetify } from 'vuetify'
import { mount, VueWrapper } from '@vue/test-utils'
import SearchInput from '@/components/SearchInput.vue'
import { beforeEach } from 'vitest'

const vuetify = createVuetify()

let wrapper: VueWrapper

beforeEach(() => {
  wrapper = mount(SearchInput, {
    props: { modelValue: 'first value' },
    global: {
      plugins: [vuetify],
    },
  })
})

describe('SearchInput', () => {
  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should change value of input if model value is changed', async () => {
    const expected = 'new value'
    const input = wrapper.find('[test-id="search-input"]')
    await input.setValue(expected)
    expect(input.attributes('value')).toBe(expected)
  })
  it('should emit "input" when changed value', () => {
    const input = wrapper.find('[test-id="search-input"]')
    input.setValue('New Test Value')
    expect(wrapper.emitted().input).toBeTruthy()
    expect(wrapper.emitted().input.length).toBe(1)
    expect(wrapper.emitted().input[1]).toEqual(undefined)
  })
  it('should emit "clear" when clicked clear icon', async () => {
    const clearIcon = wrapper.find('.v-field__clearable i')
    await clearIcon.trigger('click')

    expect(wrapper.emitted().clear).toBeTruthy()
    expect(wrapper.emitted().clear.length).toBe(1)
    expect(wrapper.emitted().clear[1]).toEqual(undefined)
  })
})
