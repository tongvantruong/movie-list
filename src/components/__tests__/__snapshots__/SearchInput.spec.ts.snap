// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchInput > should match snapshot 1`] = `
"<div data-v-54e32457="" class="search-input">
  <div data-v-54e32457="" class="v-input v-input--horizontal v-input--center-affix v-input--density-comfortable v-theme--light v-locale--is-ltr v-input--dirty v-text-field" style="max-width: 500px;" data-cy="search-input">
    <!---->
    <div class="v-input__control">
      <div class="v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--prepended v-field--single-line v-field--no-label v-field--variant-solo v-theme--light v-field--rounded v-locale--is-ltr">
        <div class="v-field__overlay"></div>
        <div class="v-field__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background"></div>
            <div class="v-progress-linear__buffer" style="width: 0%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <div class="v-field__prepend-inner"><i class="mdi-magnify mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i>
          <!---->
        </div>
        <div class="v-field__field" data-no-activator="">
          <!---->
          <!---->
          <!----><input placeholder="Search by Title" size="1" type="text" id="input-0" aria-describedby="input-0-messages" test-id="search-input" class="v-field__input" value="first value">
          <!---->
        </div>
        <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
          <div class="v-field__clearable"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear "></i></div>
        </transition-stub>
        <div class="v-field__append-inner">
          <!---->
          <!---->
        </div>
        <div class="v-field__outline">
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</div>"
`;
